datacube_stats/statistics.py

```diff
@@ -628,6 +628,136 @@ def __init__(self):
         super(MedoidNoProv, self).__init__(stat_func=_compute_medoid)
 
 
+class FCMedoid(Statistic):
```

Comment:
Nothing FC specific about this class. Needs better name. It's really just a generalisation of 'Medoid' to support `nodata` and field selection.